web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImpl.java

```diff
@@ -40,20 +41,21 @@ public FilterSickLeavesImpl(DiagnosisChapterService diagnosisChapterService, Cal
     }
 
     @Override
-    public List<SjukfallEnhet> filter(List<SjukfallEnhet> sickLeaveList, Integer fromSickLeaveLength, Integer toSickLeaveLength,
+    public List<SjukfallEnhet> filter(List<SjukfallEnhet> sickLeaveList, List<SickLeaveLengthInterval> sickLeaveLengthIntervals,
         List<DiagnosKapitel> diagnosisChapters, Integer fromPatientAge, Integer toPatientAge) {
         return sickLeaveList.stream()
-            .filter(sickLeave -> filterOnSickLeaveLength(sickLeave, fromSickLeaveLength, toSickLeaveLength))
+            .filter(sickLeave -> filterOnSickLeaveLengthIntervals(sickLeave, sickLeaveLengthIntervals))
             .filter(sickLeave -> filterOnDiagnosisChapters(sickLeave, diagnosisChapters))
             .filter(sickLeave -> filterOnPatientAge(sickLeave, fromPatientAge, toPatientAge))
             .collect(Collectors.toList());
     }
 
-    private boolean filterOnSickLeaveLength(SjukfallEnhet sickLeave, Integer fromSickLeaveLength, Integer toSickLeaveLength) {
-        if (fromSickLeaveLength == null || toSickLeaveLength == null) {
+    private boolean filterOnSickLeaveLengthIntervals(SjukfallEnhet sickLeave, List<SickLeaveLengthInterval> sickLeaveLengthIntervals) {
+        if (sickLeaveLengthIntervals == null || sickLeaveLengthIntervals.isEmpty()) {
             return true;
         }
-        return sickLeave.getDagar() >= fromSickLeaveLength && sickLeave.getDagar() <= toSickLeaveLength;
+        return sickLeaveLengthIntervals.stream()
+            .anyMatch(interval -> interval.getFrom() <= sickLeave.getDagar() && interval.getTo() >= sickLeave.getDagar());
```

Comment:
If from or to is null then the filter should allow everything over / under that date. Is that how it behaves now?

Reply:
My bad, i was not aware of that from/to could be null. Updated!